File: dataset/output/MISTRAL_codestral-latest/Task23_MISTRAL_codestral-latest.java
public class BuildingCubes {
    public static long findNb(long m) {
        long n = 1;
        while (n * n * n < m) {
            m -= n * n * n;
            n++;
        }
        return (m == n * n * n) ? n : -1;
    }
}

File: dataset/output/MISTRAL_codestral-latest/Task23_MISTRAL_codestral-latest.cpp
#include <cmath>

long long findNb(long long m) {
    long long n = 1;
    while (std::pow(n, 3) < m) {
        m -= std::pow(n, 3);
        n++;
    }
    return (m == std::pow(n, 3)) ? n : -1;
}